src/bug-filing/common/get-issue-details-html.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { escape } from 'lodash';
+import { EnvironmentInfo } from '../../common/environment-info-provider';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { title } from '../../content/strings/application';
+import { IssueUrlCreationUtils } from './issue-filing-url-string-utils';
+
+const buildBodySection = (headingHtml: string, contentHtml: string): string => {
+    return `<div><b>${headingHtml}</b></div><div>${contentHtml}</div><br>`;
+};
+
+const buildIssueDetails = (help: string, helpUrl: string, ruleId: string): string => {
+    const helpEscapedForHtml = escape(help);
+    const ruleIdEscapedForHtml = escape(ruleId);
+
+    return `${helpEscapedForHtml}<br><a href="${helpUrl}">${ruleIdEscapedForHtml}</a>`;
+};
+
+const buildApplicationHtml = (pageTitle: string, pageUrl: string): string => {
+    const pageTitleEscapedForHtml = escape(pageTitle);
+    const pageUrlEscapedForUrl = encodeURI(pageUrl);
+    const pageUrlEscapedForHtml = escape(pageUrl);
+
+    return `<a href="${pageUrlEscapedForUrl}">${pageTitleEscapedForHtml}<br>${pageUrlEscapedForHtml}</a>`;
+};
+
+const buildSnippetHtml = (snippet: string): string => {
+    const maxSnippetLength = 256;
+    let constrainedSnippet = snippet;
+
+    if (snippet.length > maxSnippetLength) {
+        constrainedSnippet = snippet.substr(0, maxSnippetLength) + '...';
+    }
+
+    return escape(constrainedSnippet);
+};
+
+const buildHowToFixHtml = (failureSummary: string): string => {
+    return escape(failureSummary)
+        .replace(/\n  /g, '<br>- ')
+        .replace(/\n /g, '<br> ')
+        .replace(/\n/g, '<br>');
+};
+
+export const getIssueDetailsHtml = (
```

Comment:
I would be tempted to explicitly declare 'getIssueDetailsHtml' as a `IssueDetailsGetter` since it's intended to be used as one.

(Parenthetical thought: after years of naming classes as nouns and methods as verbs, it feels weird to pick either part of speech as the type name for a function.)

Reply:
The general idea was to let the user of this function to enforce the type. I added though